bin/desi_tsnr_afterburner

```diff
@@ -291,192 +291,205 @@ def update_table(table1,table2,keys) :
-
-
-    #- specify names to preserve column order
-    masked_cam_summary = Table(rows=summary_rows, names=colnames)
-    for col in masked_cam_summary.colnames:
-        if isinstance(masked_cam_summary[col],Table.MaskedColumn):
-            if col in ['NIGHT', 'EXPID', 'TILEID']:
-                log.warning(f'Column {col} identified as masked column, filling masked rows with value 0 .')
-                masked_cam_summary[col].set_fill_value(0)
-            elif col in ['SURVEY', 'FAPRGRM', 'FAFLAVOR', 'GOALTYPE']:
-                log.warning(f'Column {col} identified as masked column, filling masked rows with value "unknown".')
-                masked_cam_summary[col].set_fill_value('unknown')
-            else:
-                log.warning(f'Column {col} identified as masked column, filling masked rows with value 0.0.')
-                masked_cam_summary[col].set_fill_value(0.0)
-    cam_summary = masked_cam_summary.filled()
-
-    cam_summary.meta['EXTNAME'] = 'FRAMES'
-
-    tsnr2_colnames = [x for x in cam_summary.colnames if x.startswith('TSNR2_')]
-
-    #- which rows go with which petals:
-    ispetal = dict()
-    for petal in range(10):
-        ii  = (cam_summary['CAMERA'] == 'b'+str(petal))
-        ii |= (cam_summary['CAMERA'] == 'r'+str(petal))
-        ii |= (cam_summary['CAMERA'] == 'z'+str(petal))
-        ispetal[petal] = ii
-
-    #- Distill into exposure summary per-petal
-    rows = list()
-
-    #- The unique key is the exposure Id
-    for expid in np.unique(cam_summary['EXPID'] ) :
-        ii = (cam_summary['EXPID'] == expid)
-
-        row = dict()
-        for k in ["NIGHT","EXPID","TILEID","TILERA","TILEDEC","MJD",
-                  "AIRMASS","EBV","SEEING_ETC","EFFTIME_ETC","SURVEY","GOALTYPE",
-                  "MINTFRAC","FAPRGRM","FAFLAVOR","GOALTIME"] :
-            row[k]=cam_summary[k][ii][0]
-        # mean
-        row["EXPTIME"]=float(np.mean(cam_summary['EXPTIME'][ii]))
-
-        #- Per EXPID TSNR2 is summed over cameras, averaged over petals
-        for colname in tsnr2_colnames:
-            tsnr2_petal = list()  #- TSNR^2 summed over cameras, per petal
-            for petal in range(10):
-                jj = ii & ispetal[petal]
-                if np.any(jj):
-                    tsnr2_petal.append(np.sum(cam_summary[colname][jj]))
-            row[colname] = np.mean(tsnr2_petal) # mean of petals (each being median of valid fibers)
-
-        rows.append(row)
-
-    colnames = list(rows[0].keys())
-    exp_summary = Table(rows=rows, names=colnames)
-    exp_summary.meta['EXTNAME'] = 'EXPOSURES'
+    if len(summary_rows) == 0 and preexisting_tsnr2_expid_table is not None\
+       and preexisting_tsnr2_frame_table is not None:
+        cam_summary = Table(names=preexisting_tsnr2_frame_table.colnames,
+                            dtype=preexisting_tsnr2_frame_table.dtype)
+        cam_summary.meta['EXTNAME'] = 'FRAMES'
+        exp_summary = Table(names=preexisting_tsnr2_expid_table.colnames,
```

Comment:
When `add_badexp==True`, `exp_summary` is combined with `preexisting_tsnr2_expid_table` using the `astropy.Table.vstack` method when `update_table` is called, currently on line 545 (532) in this branch (main) below:
```
 exp_summary = update_table(preexisting_tsnr2_expid_table,exp_summary,["EXPID"])
```

In the main branch, column entries that are in `preexisting_tsnr2_expid_table` but NOT in `exp_summary`, such as `SKY_MAG_G_SPEC`, `SKY_MAG_R_SPEC`, and `SKY_MAG_Z_SPEC` (because they were not present in `entry`), are masked by the `astropy.table.vstack` method, and consequently "nan" is written in the output exposures csv file for bad exposures.

In this branch, however, whenever there is no good exposure in a night to begin with, `len(summary_rows)==0`, and this new block of code is executed instead, so *by definition* all the columns in `preexisting_tsnr2_expid_table` are present in `exp_summary`. Consequently, when `update_table` is called below, values in columns such as `SKY_MAG_G_SPEC`, `SKY_MAG_R_SPEC`, and `SKY_MAG_Z_SPEC` are zero, instead of "nan". 

This may or may not be an issue that requires further changes but, if it is, the information above may be helpful in resolving it.